src/eureka/S5_lightcurve_fitting/models/FleckModel.py

```diff
@@ -0,0 +1,245 @@
+
+        log = kwargs.get('log')
+
+        # Store the ld_profile
+        self.ld_from_S4 = kwargs.get('ld_from_S4')
+        ld_func = ld_profile(self.parameters.limb_dark.value, 
+                             use_gen_ld=self.ld_from_S4)
+        len_params = len(inspect.signature(ld_func).parameters)
+        self.coeffs = ['u{}'.format(n) for n in range(len_params)[1:]]
+
+        self.ld_from_file = kwargs.get('ld_from_file')
+        
+        # Replace u parameters with generated limb-darkening values
+        if self.ld_from_S4 or self.ld_from_file:
+            log.writelog("Using the following limb-darkening values:")
+            self.ld_array = kwargs.get('ld_coeffs')
+            for c in range(self.nchannel_fitted):
+                chan = self.fitted_channels[c]
+                if self.ld_from_S4:
+                    ld_array = self.ld_array[len_params-2]
+                else:
+                    ld_array = self.ld_array
+                for u in self.coeffs:
+                    index = np.where(np.array(self.paramtitles) == u)[0]
+                    if len(index) != 0:
+                        item = self.longparamlist[c][index[0]]
+                        param = int(item.split('_')[0][-1])
+                        ld_val = ld_array[chan][param-1]
+                        log.writelog(f"{item}, {ld_val}")
+                        # Use the file value as the starting guess
+                        self.parameters.dict[item][0] = ld_val
+                        # In a normal prior, center at the file value
+                        if (self.parameters.dict[item][-1] == 'N' and
+                                self.recenter_ld_prior):
+                            self.parameters.dict[item][-3] = ld_val
+                        # Update the non-dictionary form as well
+                        setattr(self.parameters, item,
+                                self.parameters.dict[item])                     
+
+    def eval(self, channel=None, pid=None, **kwargs):
+        """Evaluate the function with the given values.
+
+        Parameters
+        ----------
+        channel : int; optional
+            If not None, only consider one of the channels. Defaults to None.
+        pid : int; optional
+            Planet ID, default is None which combines the eclipse models from
+            all planets.
+        **kwargs : dict
+            Must pass in the time array here if not already set.
+
+        Returns
+        -------
+        lcfinal : ndarray
+            The value of the model at the times self.time.
+        """
+        if channel is None:
+            nchan = self.nchannel_fitted
+            channels = self.fitted_channels
+        else:
+            nchan = 1
+            channels = [channel, ]
+            
+        if pid is None or pid == 0:
+            pid_iter = range(self.num_planets)
+            pid_id = ''
+        else:
+            pid_iter = [pid,]
+            pid_id = f'_pl{pid}'
```

Comment:
`pid_id` should not be set here and should instead be set down below and updated for each `pid` in `pid_iter`. Look at the `BatmanTransitModel` as an example

Reply:
unless I'm missing something pid_id is never set in the BatmanTransitModel? I need it here since I have to read in additional parameters, whereas BatmanTransitModel just uses AstroModel's PlanetParams which does set the pid_id

Reply:
Do you mean just move setting the pid_id inside the for `pid in pid_iter` loop?

Reply:
Ah yeah, I just meant that the `pid_id` never gets updated in the current version of your code. Thinking about it more clearly though, you shouldn't be using `pid_id` at all since the spot properties should not change between different planets. If the multi-planet fitting code is automatically making `_pl#` versions of these spot parameters, let me know I can help avoid that. I guess similarly, I should check that multi-planet fits don't duplicate the Lorentzian or Quasi-Periodic Pulsation parameters.